### ISSUE TITLE
(Linux) Fedora debug paths bundled with the ivpn-ui package cause a conflict

### ISSUE THREAD

**Issue by stenya** [ivpn/desktop-app#22]
# Bug report
A customer reported a conflict with the IVPN App and the 1Password Linux client beta on Fedora 33.

https://support.1password.com/getting-started-linux/

I have replicated the issue and the cause seems to be related to unique debug paths and symlinks in `/usr/lib/.build-id/` that are included in the `ivpn-ui` and 1Password RPM packages.

The `/usr/lib/.build-id/` paths are symlinks are supposed to be unique, so it is unclear why they are included.  It may be possible to disable debugging:
https://access.redhat.com/discussions/5045161#comment-1802881

Thanks.

**Environment**
Fedora 33

**Steps To Reproduce**
- install ivpn-ui
- install 1Password beta
- 1Password install fails due to /usr/lib/.build-id unique path conflicts

```
[j@localhost ~]$ sudo dnf install 1password
1Password                                       667  B/s | 833  B     00:01    
1Password                                       8.5 kB/s | 4.8 kB     00:00    
Importing GPG key ***:
 Userid     : "Code signing for 1Password <codesign@1password.com>"
 Fingerprint: ****
 From       : https://downloads.1password.com/linux/keys/1password.asc
Is this ok [y/N]: y
1Password                                       1.1 kB/s | 1.9 kB     00:01    
Dependencies resolved.
================================================================================
 Package          Architecture  Version                  Repository        Size
================================================================================
Installing:
 1password        x86_64        8.0.33_53.BETA-1         1password         66 M

Transaction Summary
================================================================================
Install  1 Package

Total download size: 66 M
Installed size: 244 M
Is this ok [y/N]: y
Downloading Packages:
1password-8.0.33-53.BETA.x86_64.rpm                                                                                           5.1 MB/s |  66 MB     00:13    
--------------------------------------------------------------------------------------------------------------------------------------------------------------
Total                                                                                                                         5.1 MB/s |  66 MB     00:13     
Running transaction check
Transaction check succeeded.
Running transaction test
The downloaded packages were saved in cache until the next successful transaction.
You can remove cached packages by executing 'dnf clean packages'.
Error: Transaction test error:
  file /usr/lib/.build-id/13/070cd6d4401903dee7a42e402d150de4362b18 from install of 1password-8.0.33_53.BETA-1.x86_64 conflicts with file from package ivpn-ui-3.3.10-1.x86_64
  file /usr/lib/.build-id/3e/d503d7be965ab698b328fe3a69162a4efb5351 from install of 1password-8.0.33_53.BETA-1.x86_64 conflicts with file from package ivpn-ui-3.3.10-1.x86_64
  file /usr/lib/.build-id/50/300d22dcba520f8681cbf4f8bd71ff9320371c from install of 1password-8.0.33_53.BETA-1.x86_64 conflicts with file from package ivpn-ui-3.3.10-1.x86_64
  file /usr/lib/.build-id/5b/8864b69ddfe3bd034d5dbef9b814a8f9b59cbd from install of 1password-8.0.33_53.BETA-1.x86_64 conflicts with file from package ivpn-ui-3.3.10-1.x86_64
  file /usr/lib/.build-id/ae/f8abf3f50b39abd59d73628e8cd89cceee8f54 from install of 1password-8.0.33_53.BETA-1.x86_64 conflicts with file from package ivpn-ui-3.3.10-1.x86_64
  file /usr/lib/.build-id/af/4374e54497edac9a84703037832fafb0d99308 from install of 1password-8.0.33_53.BETA-1.x86_64 conflicts with file from package ivpn-ui-3.3.10-1.x86_64
  file /usr/lib/.build-id/b4/fe1769dc0dbcd634d93f5ffd826d0354016f9e from install of 1password-8.0.33_53.BETA-1.x86_64 conflicts with file from package ivpn-ui-3.3.10-1.x86_64
  file /usr/lib/.build-id/f0/d8c634c19103e2e50b980cc1fb5cbacdc8d07c from install of 1password-8.0.33_53.BETA-1.x86_64 conflicts with file from package ivpn-ui-3.3.10-1.x86_64
  file /usr/lib/.build-id/f3/13677b4c4d959f49800be71689dc788f06ae94 from install of 1password-8.0.33_53.BETA-1.x86_64 conflicts with file from package ivpn-ui-3.3.10-1.x86_64

[j@localhost ~]$



[root@localhost ~]# ls -l /usr/lib/.build-id/13/070cd6d4401903dee7a42e402d150de4362b18
lrwxrwxrwx. 1 root root 40 Apr 23 12:19 /usr/lib/.build-id/13/070cd6d4401903dee7a42e402d150de4362b18 -> ../../../../opt/ivpn/ui/bin/libffmpeg.so
[root@localhost ~]# ls -l /usr/lib/.build-id/3e/d503d7be965ab698b328fe3a69162a4efb5351
lrwxrwxrwx. 1 root root 49 Apr 23 12:19 /usr/lib/.build-id/3e/d503d7be965ab698b328fe3a69162a4efb5351 -> ../../../../opt/ivpn/ui/bin/swiftshader/libEGL.so
[root@localhost ~]# ls -l /usr/lib/.build-id/50/300d22dcba520f8681cbf4f8bd71ff9320371c
lrwxrwxrwx. 1 root root 35 Apr 23 12:19 /usr/lib/.build-id/50/300d22dcba520f8681cbf4f8bd71ff9320371c -> ../../../../opt/ivpn/ui/bin/ivpn-ui
[root@localhost ~]# ls -l /usr/lib/.build-id/5b/8864b69ddfe3bd034d5dbef9b814a8f9b59cbd
lrwxrwxrwx. 1 root root 52 Apr 23 12:19 /usr/lib/.build-id/5b/8864b69ddfe3bd034d5dbef9b814a8f9b59cbd -> ../../../../opt/ivpn/ui/bin/swiftshader/libGLESv2.so
[root@localhost ~]# ls -l /usr/lib/.build-id/ae/f8abf3f50b39abd59d73628e8cd89cceee8f54
lrwxrwxrwx. 1 root root 40 Apr 23 12:19 /usr/lib/.build-id/ae/f8abf3f50b39abd59d73628e8cd89cceee8f54 -> ../../../../opt/ivpn/ui/bin/libGLESv2.so
[root@localhost ~]# ls -l /usr/lib/.build-id/af/4374e54497edac9a84703037832fafb0d99308
lrwxrwxrwx. 1 root root 37 Apr 23 12:19 /usr/lib/.build-id/af/4374e54497edac9a84703037832fafb0d99308 -> ../../../../opt/ivpn/ui/bin/libEGL.so
[root@localhost ~]# ls -l /usr/lib/.build-id/b4/fe1769dc0dbcd634d93f5ffd826d0354016f9e
lrwxrwxrwx. 1 root root 42 Apr 23 12:19 /usr/lib/.build-id/b4/fe1769dc0dbcd634d93f5ffd826d0354016f9e -> ../../../../opt/ivpn/ui/bin/chrome-sandbox
[root@localhost ~]# ls -l /usr/lib/.build-id/f0/d8c634c19103e2e50b980cc1fb5cbacdc8d07c
lrwxrwxrwx. 1 root root 42 Apr 23 12:19 /usr/lib/.build-id/f0/d8c634c19103e2e50b980cc1fb5cbacdc8d07c -> ../../../../opt/ivpn/ui/bin/libvulkan.so.1
[root@localhost ~]# ls -l /usr/lib/.build-id/f3/13677b4c4d959f49800be71689dc788f06ae94
lrwxrwxrwx. 1 root root 48 Apr 23 12:19 /usr/lib/.build-id/f3/13677b4c4d959f49800be71689dc788f06ae94 -> ../../../../opt/ivpn/ui/bin/libvk_swiftshader.so
Checking the RPM packages shows that both `ivpn-ui-3.3.10-1.x86_64.rpm` and `1password-8.0.33-53.BETA.x86_64.rpm` include the same unique `/usr/lib/.build-id/` paths and files:

[j@localhost ~]$ rpm -qlp ivpn-ui-3.3.10-1.x86_64.rpm
/opt/ivpn/ui/IVPN.desktop
/opt/ivpn/ui/bin/LICENSE.electron.txt
/opt/ivpn/ui/bin/LICENSES.chromium.html
/opt/ivpn/ui/bin/chrome-sandbox
/opt/ivpn/ui/bin/chrome_100_percent.pak
/opt/ivpn/ui/bin/chrome_200_percent.pak
/opt/ivpn/ui/bin/icudtl.dat
/opt/ivpn/ui/bin/ivpn-ui
/opt/ivpn/ui/bin/libEGL.so
/opt/ivpn/ui/bin/libGLESv2.so
/opt/ivpn/ui/bin/libffmpeg.so
/opt/ivpn/ui/bin/libvk_swiftshader.so
/opt/ivpn/ui/bin/libvulkan.so.1
/opt/ivpn/ui/bin/locales/am.pak
...
/opt/ivpn/ui/bin/locales/zh-TW.pak
/opt/ivpn/ui/bin/resources.pak
/opt/ivpn/ui/bin/resources/app.asar
/opt/ivpn/ui/bin/resources/public.pem
/opt/ivpn/ui/bin/snapshot_blob.bin
/opt/ivpn/ui/bin/swiftshader/libEGL.so
/opt/ivpn/ui/bin/swiftshader/libGLESv2.so
/opt/ivpn/ui/bin/v8_context_snapshot.bin
/opt/ivpn/ui/bin/vk_swiftshader_icd.json
/opt/ivpn/ui/ivpnicon.svg
/usr/lib/.build-id
/usr/lib/.build-id/13
/usr/lib/.build-id/13/070cd6d4401903dee7a42e402d150de4362b18
/usr/lib/.build-id/3e
/usr/lib/.build-id/3e/d503d7be965ab698b328fe3a69162a4efb5351
/usr/lib/.build-id/50
/usr/lib/.build-id/50/300d22dcba520f8681cbf4f8bd71ff9320371c
/usr/lib/.build-id/5b
/usr/lib/.build-id/5b/8864b69ddfe3bd034d5dbef9b814a8f9b59cbd
/usr/lib/.build-id/ae
/usr/lib/.build-id/ae/f8abf3f50b39abd59d73628e8cd89cceee8f54
/usr/lib/.build-id/af
/usr/lib/.build-id/af/4374e54497edac9a84703037832fafb0d99308
/usr/lib/.build-id/b4
/usr/lib/.build-id/b4/fe1769dc0dbcd634d93f5ffd826d0354016f9e
/usr/lib/.build-id/f0
/usr/lib/.build-id/f0/d8c634c19103e2e50b980cc1fb5cbacdc8d07c
/usr/lib/.build-id/f3
/usr/lib/.build-id/f3/13677b4c4d959f49800be71689dc788f06ae94
[root@localhost 1p]# rpm -qlp 1password-8.0.33-53.BETA.x86_64.rpm 
/opt/1Password/1Password-BrowserSupport
/opt/1Password/1Password-KeyringHelper
/opt/1Password/1password
/opt/1Password/LICENSE.electron.txt
/opt/1Password/LICENSES.chromium.html
/opt/1Password/after-install.sh
/opt/1Password/after-remove.sh
/opt/1Password/chrome-sandbox
/opt/1Password/chrome_100_percent.pak
/opt/1Password/chrome_200_percent.pak
/opt/1Password/com.1password.1Password.policy
/opt/1Password/icudtl.dat
/opt/1Password/install_biometrics_policy.sh
/opt/1Password/libEGL.so
/opt/1Password/libGLESv2.so
/opt/1Password/libffmpeg.so
/opt/1Password/libvk_swiftshader.so
/opt/1Password/libvulkan.so.1
/opt/1Password/locales/am.pak
...
/opt/1Password/locales/zh-TW.pak
/opt/1Password/resources.pak
/opt/1Password/resources/1password.desktop
/opt/1Password/resources/app.asar
/opt/1Password/resources/app.asar.unpacked/CREDITS.html
/opt/1Password/resources/app.asar.unpacked/index.node
/opt/1Password/resources/custom_allowed_browsers
/opt/1Password/resources/icons/hicolor/256x256/apps/1password.png
/opt/1Password/resources/icons/hicolor/32x32/apps/1password.png
/opt/1Password/resources/icons/hicolor/512x512/apps/1password.png
/opt/1Password/resources/icons/hicolor/64x64/apps/1password.png
/opt/1Password/snapshot_blob.bin
/opt/1Password/swiftshader/libEGL.so
/opt/1Password/swiftshader/libGLESv2.so
/opt/1Password/v8_context_snapshot.bin
/opt/1Password/vk_swiftshader_icd.json
/usr/lib/.build-id
/usr/lib/.build-id/13
/usr/lib/.build-id/13/070cd6d4401903dee7a42e402d150de4362b18
/usr/lib/.build-id/3e
/usr/lib/.build-id/3e/d503d7be965ab698b328fe3a69162a4efb5351
/usr/lib/.build-id/50
/usr/lib/.build-id/50/300d22dcba520f8681cbf4f8bd71ff9320371c
/usr/lib/.build-id/5b
/usr/lib/.build-id/5b/8864b69ddfe3bd034d5dbef9b814a8f9b59cbd
/usr/lib/.build-id/9d
/usr/lib/.build-id/9d/41f958929cbe6474f668de4c96432a4c99ba88
/usr/lib/.build-id/a7
/usr/lib/.build-id/a7/033af106723fab4aee68536984ae7ffc8cc522
/usr/lib/.build-id/ae
/usr/lib/.build-id/ae/f8abf3f50b39abd59d73628e8cd89cceee8f54
/usr/lib/.build-id/af
/usr/lib/.build-id/af/4374e54497edac9a84703037832fafb0d99308
/usr/lib/.build-id/b4
/usr/lib/.build-id/b4/fe1769dc0dbcd634d93f5ffd826d0354016f9e
/usr/lib/.build-id/f0
/usr/lib/.build-id/f0/d8c634c19103e2e50b980cc1fb5cbacdc8d07c
/usr/lib/.build-id/f3
/usr/lib/.build-id/f3/13677b4c4d959f49800be71689dc788f06ae94
/usr/share/applications/1password.desktop
/usr/share/icons/hicolor/256x256/apps/1password.png
/usr/share/icons/hicolor/32x32/apps/1password.png
/usr/share/icons/hicolor/512x512/apps/1password.png
/usr/share/icons/hicolor/64x64/apps/1password.png
Note: I removed most of the `/locales/` to try to keep this shorter.

The `ivpn-3.3.7-1.x86_64.rpm` does not include these paths:

$ rpm -qlp ivpn-3.3.7-1.x86_64.rpm 
/opt/ivpn/etc/ca.crt
/opt/ivpn/etc/client.down
/opt/ivpn/etc/client.up
/opt/ivpn/etc/firewall.sh
/opt/ivpn/etc/servers.json
/opt/ivpn/etc/ta.key
/usr/lib/.build-id
/usr/lib/.build-id/fc
/usr/lib/.build-id/fc/397197e3ee1aa130951239ee94cd15bc9e17f0
/usr/local/bin/ivpn
/usr/local/bin/ivpn-service
/usr/share/pleaserun/ivpn-service/generate-cleanup.sh
/usr/share/pleaserun/ivpn-service/install-path.sh
/usr/share/pleaserun/ivpn-service/install.sh
/usr/share/pleaserun/ivpn-service/launchd/10.9/files/Library/LaunchDaemons/ivpn-service.plist
/usr/share/pleaserun/ivpn-service/launchd/10.9/install_actions.sh
/usr/share/pleaserun/ivpn-service/systemd/default/files/etc/default/ivpn-service
/usr/share/pleaserun/ivpn-service/systemd/default/files/etc/systemd/system/ivpn-service.service
/usr/share/pleaserun/ivpn-service/systemd/default/install_actions.sh
/usr/share/pleaserun/ivpn-service/sysv/lsb-3.1/files/etc/default/ivpn-service
/usr/share/pleaserun/ivpn-service/sysv/lsb-3.1/files/etc/init.d/ivpn-service
/usr/share/pleaserun/ivpn-service/upstart/0.6.5/files/etc/default/ivpn-service
/usr/share/pleaserun/ivpn-service/upstart/0.6.5/files/etc/init/ivpn-service.conf
/usr/share/pleaserun/ivpn-service/upstart/1.5/files/etc/default/ivpn-service
/usr/share/pleaserun/ivpn-service/upstart/1.5/files/etc/init/ivpn-service.conf
```

**Comment by MillironX** [comment]
I get this same problem with the [FreeTube](https://github.com/FreeTubeApp/FreeTube) RPMs on Fedora 34.

```shellsession
$ sudo dnf install https://github.com/FreeTubeApp/FreeTube/releases/download/v0.13.0-beta/freetube_0.13.0_amd64.rpm -y
Last metadata expiration check: 0:34:59 ago on Wed 02 Jun 2021 09:08:13 AM CDT.
freetube_0.13.0_amd64.rpm                                                                     2.8 MB/s |  72 MB     00:25    
Dependencies resolved.
==============================================================================================================================
 Package                      Architecture               Version                       Repository                        Size
==============================================================================================================================
Installing:
 freetube                     x86_64                     0.13.0-1                      @commandline                      72 M

Transaction Summary
==============================================================================================================================
Upgrade  1 Package

Total size: 72 M
Downloading Packages:
Running transaction check
Transaction check succeeded.
Running transaction test
Error: Transaction test error:
  file /usr/lib/.build-id/13/070cd6d4401903dee7a42e402d150de4362b18 from install of freetube-0.13.0-1.x86_64 conflicts with file from package ivpn-ui-3.3.10-1.x86_64
  file /usr/lib/.build-id/3e/d503d7be965ab698b328fe3a69162a4efb5351 from install of freetube-0.13.0-1.x86_64 conflicts with file from package ivpn-ui-3.3.10-1.x86_64
  file /usr/lib/.build-id/5b/8864b69ddfe3bd034d5dbef9b814a8f9b59cbd from install of freetube-0.13.0-1.x86_64 conflicts with file from package ivpn-ui-3.3.10-1.x86_64
  file /usr/lib/.build-id/af/4374e54497edac9a84703037832fafb0d99308 from install of freetube-0.13.0-1.x86_64 conflicts with file from package ivpn-ui-3.3.10-1.x86_64
  file /usr/lib/.build-id/b4/fe1769dc0dbcd634d93f5ffd826d0354016f9e from install of freetube-0.13.0-1.x86_64 conflicts with file from package ivpn-ui-3.3.10-1.x86_64
  file /usr/lib/.build-id/f0/d8c634c19103e2e50b980cc1fb5cbacdc8d07c from install of freetube-0.13.0-1.x86_64 conflicts with file from package ivpn-ui-3.3.10-1.x86_64
  file /usr/lib/.build-id/f3/13677b4c4d959f49800be71689dc788f06ae94 from install of freetube-0.13.0-1.x86_64 conflicts with file from package ivpn-ui-3.3.10-1.x86_64
```

**Comment by gorkapernas** [comment]
Verified fixed on version 3.3.21, Fedora 33, the '.build-id' folder is no longer created during the installation, therefore the issue raised in the ticket doesn't occur anymore. Currently it is possible to install 1Password or FreeTube without any conflicting issues with IVPN.

**Comment by stenya** [comment]
Released in v3.3.30
